controllers/kafkatopic_controller.go

```diff
@@ -95,6 +106,19 @@ func (r *KafkaTopicReconciler) Reconcile(ctx context.Context, request reconcile.
 		return requeueWithError(reqLogger, "failed to lookup referenced cluster", err)
 	}
 
+	// Set managed status based on KafkaTopic managedBy annotation
+	managedByStatus := webhooks.TopicManagedByKoperatorAnnotationValue
+	if !isTopicManagedByKoperator(instance) {
+		managedByStatus = instance.GetAnnotations()[webhooks.TopicManagedByAnnotationKey]
+	}
```

Comment:
**Note**: here the logic depends on the `isTopicManagedByKoperator` function as well.

Reply:
I left this comment to have a placeholder in case we want to modify the `!ok` handling, so it would be easier to double check the cases where it is used.